sentry-spring/src/main/java/io/sentry/spring/SentrySpringFilter.java

```diff
@@ -0,0 +1,100 @@
+package io.sentry.spring;
+
+import static io.sentry.SentryOptions.RequestSize.*;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.Breadcrumb;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions.RequestSize;
+import io.sentry.util.Objects;
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.http.MediaType;
+import org.springframework.util.MimeType;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Open
+public class SentrySpringFilter extends OncePerRequestFilter {
+  private final @NotNull IHub hub;
+  private final @NotNull SentryRequestResolver requestResolver;
+
+  public SentrySpringFilter(@NotNull IHub hub, @NotNull SentryRequestResolver requestResolver) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    this.requestResolver = Objects.requireNonNull(requestResolver, "requestResolver is required");
+  }
+
+  public SentrySpringFilter(final @NotNull IHub hub) {
+    this(hub, new SentryRequestResolver(hub));
+  }
+
+  public SentrySpringFilter() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  protected void doFilterInternal(
+      final @NotNull HttpServletRequest servletRequest,
+      final @NotNull HttpServletResponse response,
+      final @NotNull FilterChain filterChain)
+      throws ServletException, IOException {
+    if (hub.isEnabled()) {
+      final HttpServletRequest request = resolveHttpServletRequest(servletRequest);
+      try {
+        hub.pushScope();
+        hub.addBreadcrumb(Breadcrumb.http(request.getRequestURI(), request.getMethod()));
```

Comment:
Might need to confirm if this looks like it's an incoming request. I believe this "format" of crumb might have been designed to represent an outgoing HTTP request. 
If it won't get confusing with crumbs representing requests done from this process out to the another for example through rest-template, then we're good.

Reply:
That's how it's been since 3.0 release. Where can I verify if this is the right format?

Reply:
If we have an incoming request into a spring process that makes an outgoing request. Would we be able to differentiate the crumbs? Probably at least through the URL.
Doesn't block this PR